src/gpodder/plugins/soundcloud.py

```diff
@@ -91,14 +91,19 @@ def get_metadata(url):
 class SoundcloudUser(object):
     def __init__(self, username):
         self.username = username
+        self.cache = {}
 
     def get_user_info(self):
         global CONSUMER_KEY
         key = ':'.join((self.username, 'user_info'))
 
+        if key in self.cache:
```

Comment:
In general I find it easier to read this way:

```
if key not in cache:
    cache[key] = ...

return cache[key]
```

This way you are always reading from the cache and can be sure you return the right element (with the current code, it could be that you accidentally return something else, although the way the code is right now it looks like all is fine). Just imagine the following:

```
if key in cache:
    return cache[key]

cache[key] = something['data']
return something
```

Was it easy to spot the bug there?

Reply:
You're returning the parent item instead of [data] but I don't know that I would have noticed it if you had not specifically asked if there was a bug there.

Point taken.